python/lancedb/remote/table.py

```diff
@@ -104,6 +108,28 @@ def create_index(
         ... )
         >>> table.create_index("L2", "vector") # doctest: +SKIP
         """
+
+        if num_partitions is not None:
+            raise NotImplementedError(
```

Comment:
Can we just `log.warning` here?  This call can get through with these parameters, don't need to interrupt user's workflow.